pkg/k8sutil/create_test.go

```diff
@@ -287,3 +288,177 @@ func TestCreateOrUpdateNamespaceUpdateSuccess(t *testing.T) {
 		t.Fatalf("expected %q, got: %q", name, mockns.ObjectMeta.Labels[internal.NamespaceLabelKey])
 	}
 }
+
+func TestYAMLToUnstructured(t *testing.T) {
+	tests := []struct {
+		name    string
+		yamlObj []byte
+		want    *unstructured.Unstructured
+		wantErr bool
+	}{
+		{
+			name: "Valid_Kubernetes_object",
+			yamlObj: []byte(`apiVersion: v1
+kind: Namespace
+metadata:
+  name: test`),
+			want: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "v1",
+					"kind":       "Namespace",
+					"metadata":   map[string]interface{}{"name": "test"},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "Invalid_Kubernetes_object",
+			yamlObj: []byte(`foobar`),
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := YAMLToUnstructured(tt.yamlObj)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("got error: %v\nwantErr: %v", err, tt.wantErr)
+
+				return
+			}
+
+			if err != nil && tt.wantErr {
+				t.Logf("successfully failed with error: %v", err)
+
+				return
```

Comment:
It this needed? It seems like a debug message and a funny one `successfully failed with error` :smile:

Reply:
Yeah so that the test author knows that it failed with the error that they wanted and not with something else.

Reply:
I think such visual inspection is a poor habit...